crates/bevy_ecs/src/system/exclusive_function_system.rs

```diff
@@ -28,18 +29,21 @@ where
     system_meta: SystemMeta,
     world_id: Option<WorldId>,
     // NOTE: PhantomData<fn()-> T> gives this safe Send/Sync impls
-    marker: PhantomData<fn() -> Marker>,
+    marker: PhantomData<fn() -> (In, Out, Marker)>,
```

Comment:
Shouldn't this be contravariant in `In` (i.e. `In` as a parameter of the `fn()`) since it's an input of the system?

Reply:
That's a good point. `In` has a `'static` bound so I'm not sure if it matters, but it certainly looks a lot better to do `fn(In) -> Out`

Reply:
AFAIK you can still have subtyping between `'static` types, e.g. between `for<'a> fn(&'a i32)` and `fn(&'static i32)`